### PR TITLE
Issue 1082 - Users tab

### LINKED ISSUE #1082: User Management: Users Tab
### Description
Merging "Licenses & Jobs" view with "Assign Permissions (teamspaces tab)" view into "Users" tab.
https://sketch.cloud/s/mrZ88/all/3d-repo-monday/users

### Goals
- [x] Admins can manage members and their permissions from single view
- [x] Admins can add new members to teamspace
- [x] Admins can change job title
- [x] Admins can change permissions
- [x] Admins can remove members from teamspace

### Tasks
- [x] New application state: User Management
- [x] Users list component
- [x] FAB for adding new users
- [x] Information about licences in tab's footer

### PR DESCRIPTION
**Pull request by simplypixi**
This fixes #1082 

#### Description

- Added new application state: User Management
- Added Users list component
- Added FAB for adding new users. This FAB opens a panel with form.
- Added information about licences in tab's footer
- Removed old Licences view
- Added "plop" library to speed up the creation of new components
- Changed logic of some backend routes and model's methods
- Added a lot of small code fixes

### CODE REVIEW

**Comment by simplypixi** [comment]
@carmenfan

> Change Email display to company (for both add new user search and the teamspace user list)

What does "company" mean in user context? I don't see that field in the model.

**Comment by carmenfan** [comment]
@simplypixi 

inside billing.billingInfo, there is a field called company. This is an optional field during signup, so if the user has it we'd like to display it, if not just leave it blank

File: backend/models/user.js [hunk 1]
```diff
@@ -146,6 +146,34 @@ schema.statics.findByUserName = function(user) {
 	return this.findOne({account: "admin"}, { user });
 };
 
+schema.statics.findAllByQuery = function (teamspace, query) {
```
[the remark this answers is not in the document]
Comment (simplypixi) [reply backend/models/user.js]: fixed

File: frontend/components/account/js/account-user-management.component.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+		public static $inject: string[] = [
+			"$q",
+			"AccountService",
+			"DialogService"
+		];
+
+		private TABS_TYPES = TABS_TYPES;
+
+		private account;
+		private accounts;
```
[the remark this answers is not in the document]
Comment (simplypixi) [reply frontend/components/account/js/account-user-management.component.ts]: fixed

File: backend/routes/teamspace.js [hunk 3]
```diff
@@ -48,10 +50,20 @@
 		}).then(quotaInfo => {
 			responseCodes.respond(utils.APIInfo(req), req, res, next, responseCodes.OK, quotaInfo);
 		}).catch(err => {
-
 			responseCodes.respond(utils.APIInfo(req), req, res, next, err, err);
 		});
+	}
 
+	function findUsersWithoutMembership(req, res, next) {
+		User.findAllByQuery(req.params.account, req.params.query).then((users) => {
```
[the remark this answers is not in the document]
Comment (simplypixi) [reply backend/routes/teamspace.js]: fixed

File: backend/models/user.js [hunk 4]
```diff
@@ -1002,26 +1030,51 @@ schema.methods.removeTeamMember = function(username, cascadeRemove) {
...
+					return Promise.all(promises).then(userEntry.getBaseData.bind(userEntry));
+				})
+					.then(userData => Object.assign({job, permissions}, userData));
+
 			});
 		}
 	});
 };
 
+schema.methods.getBaseData = function() {
```
[the remark this answers is not in the document]
Comment (simplypixi) [reply backend/models/user.js]: fixed

File: backend/models/user.js [hunk 5]
```diff
@@ -1002,26 +1030,51 @@ schema.methods.removeTeamMember = function(username, cascadeRemove) {
...
+schema.methods.addTeamMember = function(user, job, permissions) {
 	return User.getAllUsersInTeamspace(this.user).then((userArr) => {
 		const limits = this.customData.billing.getSubscriptionLimits();
 		if(limits.collaboratorLimit !== "unlimited" && userArr.length >= limits.collaboratorLimit) {
 			return Promise.reject(responseCodes.LICENCE_LIMIT_REACHED);
 		} else {
-			return User.findByUserName(user).then(userEntry => {
+			return Promise.all([
+				User.findByUserName(user),
+				User.findByUserName(this.user)
```
[the remark this answers is not in the document]
Comment (simplypixi) [reply backend/models/user.js]: fixed

**Pull request merged**